### DISCUSSION TITLE
Debug git config used by gitui

### DISCUSSION THREAD

**Discussion by Abdillah** [extrawurst/gitui#1126] [Q&A]
Hello, I tried `gitui` but failed to commit due to this error.

```
00:03:22 [ERROR] commit error: git error:config value 'user.name' was not found; class=Config (7); code=NotFound (-3)
```

...although git commit is working fine. I have all config necessary to commit but on global level i.e. `git config --local --list` doesn't have "user.name". I want to debug this as end user and make sure this is the case, **how to do that?**

It's possible that because I use different config file location, the global config is not recognized by gitui (or libgit2?).

**Comment by extrawurst** [comment]
Not sure if this is what you mean but here you can see how we read the username from the config: https://github.com/extrawurst/gitui/blob/master/asyncgit/src/sync/commit.rs#L50

should be easy to build a simple repro of this in git2-rs or libgit2 against your custom setup.

sounds a lot like something libgit2 needs to implement
